File: src/test/java/io/georocket/util/ThrowableHelperTest.java
package io.georocket.util;

import static org.junit.Assert.assertEquals;

import java.io.FileNotFoundException;

import org.junit.Test;

import io.vertx.core.eventbus.ReplyException;
import io.vertx.core.eventbus.ReplyFailure;

/**
 * Test {@link ThrowableHelper}
 * @author Andrej Sajenko
 */
public class ThrowableHelperTest {
  /**
   * Test code of {@link ReplyException}
   */
  @Test
  public void testThrowableToCodeReplyException() {
    int expectedCode = 505;
    
    Throwable throwable = new ReplyException(ReplyFailure.NO_HANDLERS, expectedCode, "Message");
    
    int statusCode = ThrowableHelper.throwableToCode(throwable);
    assertEquals(expectedCode, statusCode);
  }

  /**
   * Test code of {@link IllegalArgumentException}
   */
  @Test
  public void testThrowableToCodeIllegalArgument() {
    int expectedCode = 400;
    
    Throwable throwable = new IllegalArgumentException();
    
    int statusCode = ThrowableHelper.throwableToCode(throwable);
    assertEquals(expectedCode, statusCode);
  }

  /**
   * Test code of {@link FileNotFoundException}
   */
  @Test
  public void testThrowableToCodeFileNotFound() {
    int expectedCode = 404;
    
    Throwable throwable = new FileNotFoundException();
    
    int statusCode = ThrowableHelper.throwableToCode(throwable);
    assertEquals(expectedCode, statusCode);
  }

  /**
   * Test code of unknown throwables.
   */
  @Test
  public void testThrowableToCodeThrowable() {
    int expectedCode = 500;
    
    Throwable throwable = new Throwable();
    
    int statusCode = ThrowableHelper.throwableToCode(throwable);
    assertEquals(expectedCode, statusCode);
  }

  /**
   * Test throwable message
   */
  @Test
  public void testThrowableToMessage() {
    String expectedMessage = "A Error happen!";
    String defaultMessage = "Oops!";
    
    Throwable throwable = new Throwable(expectedMessage);
    
    String message = ThrowableHelper.throwableToMessage(throwable, defaultMessage);
    assertEquals(expectedMessage, message);
  }

  /**
   * Test default message
   */
  @Test
  public void testThrowableToMessageDefault() {
    String defaultMessage = "Oops!";
    
    Throwable throwable = new Throwable();
    
    String message = ThrowableHelper.throwableToMessage(throwable, defaultMessage);
    assertEquals(defaultMessage, message);
  }
}

File: src/test/kotlin/io/georocket/util/ThrowableHelperTest.kt
package io.georocket.util

import io.georocket.util.ThrowableHelper.throwableToCode
import io.georocket.util.ThrowableHelper.throwableToMessage
import io.vertx.core.eventbus.ReplyException
import io.vertx.core.eventbus.ReplyFailure
import org.junit.Assert
import org.junit.Test
import java.io.FileNotFoundException

/**
 * Test [ThrowableHelper]
 * @author Andrej Sajenko
 */
class ThrowableHelperTest {
  /**
   * Test code of [ReplyException]
   */
  @Test
  fun testThrowableToCodeReplyException() {
    val expectedCode = 505
    val throwable: Throwable = ReplyException(ReplyFailure.NO_HANDLERS, expectedCode, "Message")
    val statusCode = throwableToCode(throwable)
    Assert.assertEquals(expectedCode.toLong(), statusCode.toLong())
  }

  /**
   * Test code of [IllegalArgumentException]
   */
  @Test
  fun testThrowableToCodeIllegalArgument() {
    val expectedCode = 400
    val throwable: Throwable = IllegalArgumentException()
    val statusCode = throwableToCode(throwable)
    Assert.assertEquals(expectedCode.toLong(), statusCode.toLong())
  }

  /**
   * Test code of [FileNotFoundException]
   */
  @Test
  fun testThrowableToCodeFileNotFound() {
    val expectedCode = 404
    val throwable: Throwable = FileNotFoundException()
    val statusCode = throwableToCode(throwable)
    Assert.assertEquals(expectedCode.toLong(), statusCode.toLong())
  }

  /**
   * Test code of unknown throwables.
   */
  @Test
  fun testThrowableToCodeThrowable() {
    val expectedCode = 500
    val throwable = Throwable()
    val statusCode = throwableToCode(throwable)
    Assert.assertEquals(expectedCode.toLong(), statusCode.toLong())
  }

  /**
   * Test throwable message
   */
  @Test
  fun testThrowableToMessage() {
    val expectedMessage = "A Error happen!"
    val defaultMessage = "Oops!"
    val throwable = Throwable(expectedMessage)
    val message = throwableToMessage(throwable, defaultMessage)
    Assert.assertEquals(expectedMessage, message)
  }

  /**
   * Test default message
   */
  @Test
  fun testThrowableToMessageDefault() {
    val defaultMessage = "Oops!"
    val throwable = Throwable()
    val message = throwableToMessage(throwable, defaultMessage)
    Assert.assertEquals(defaultMessage, message)
  }
}
